src/bootstrap/src/core/config/config.rs

```diff
@@ -2100,6 +2091,94 @@ impl Config {
 
         Some(commit.to_string())
     }
+
+    fn parse_download_ci_llvm(
+        &self,
+        download_ci_llvm: Option<StringOrBool>,
+        asserts: bool,
+    ) -> bool {
+        match download_ci_llvm {
+            None => self.channel == "dev" && llvm::is_ci_llvm_available(&self, asserts),
+            Some(StringOrBool::Bool(b)) => b,
+            Some(StringOrBool::String(s)) if s == "if-available" => {
+                llvm::is_ci_llvm_available(&self, asserts)
+            }
+            Some(StringOrBool::String(s)) if s == "if-unchanged" => {
+                if !self.rust_info.is_from_tarball() {
+                    if self
+                        .last_modified_commit(&["src/llvm-project"], "download-ci-llvm", true)
+                        .is_none()
+                    {
+                        // there are some untracked changes in the the given paths.
+                        false
+                    } else {
+                        llvm::is_ci_llvm_available(&self, asserts)
+                    }
+                } else {
+                    llvm::is_ci_llvm_available(&self, asserts)
+                }
```

Comment:
Both 'if' statements appear to be combined, so we can avoid duplicating the 'else' clause.